irohad/main/irohad.cpp

```diff
@@ -159,6 +172,62 @@ static shared_model::crypto::Keypair getKeypairFromFile(
              });
 }
 
+boost::optional<std::string> getEnvVar(std::string const& key)
+{
+    char const* val = getenv(key.c_str()); 
+    return val == NULL ? boost::optional<std::string>() : std::string(val);
+}
+
+boost::optional<IrohadConfig::DbConfig> getPostgresCredsFromEnv() {
+  
+  auto pg_host = getEnvVar("IROHA_POSTGRES_HOST");
+  auto pg_port_str = getEnvVar("IROHA_POSTGRES_PORT");
+  auto pg_user = getEnvVar("IROHA_POSTGRES_USER");
+  auto pg_pass = getEnvVar("IROHA_POSTGRES_PASSWORD");
+  auto pg_w_dbname = getEnvVar("IROHA_POSTGRES_DATABASE");
+  auto pg_m_dbname = getEnvVar("IROHA_POSTGRES_MAINTENANCE_DATABASE");
+
+  std::uint16_t pg_port;
+  if (pg_port_str.has_value()) {
+    auto pg_port_int = std::stoi(pg_port_str.value());
+    if (pg_port_int < std::numeric_limits<std::uint16_t>::min() ||
+        pg_port_int > std::numeric_limits<std::uint16_t>::max()) {
```

Comment:
Shouldn't be like that:
```suggestion
    if (pg_port_int <= std::numeric_limits<std::uint16_t>::min() ||
        pg_port_int >= std::numeric_limits<std::uint16_t>::max()) {
```
because maximum and minimum values are also acceptable, aren't they?

Reply:
@baziorek Agreed that the full uint16 range is acceptable, but confused also, because this if condition is throwing if the port is out of range, e.g. it's checking to see if the port is invalid, not checking if the port is valid. Based on that, changing the current logic to to make it range-inclusive would make it crash if one specifies the postgres port as `2^16-1` (65535) for example.

Reply:
@baziorek I ended up using the shortcut from @kuvaldini once I saw that